db/db_iter.cc

```diff
@@ -70,6 +73,23 @@ class DBIter: public Iterator {
     assert(valid_);
     return (direction_ == kForward) ? iter_->value() : saved_value_;
   }
+  // Riak specific:  if a database iterator, returns key meta data
+  // REQUIRES: Valid() and forward iteration
+  //  (reverse iteration is possible, just needs code)
+  virtual KeyMetaData & keymetadata() const
+  {
+    assert(valid_ && kForward==direction_);
+    if (kForward==direction_)
+    {
+      ParsedInternalKey parsed;
+      ParseInternalKey(iter_->key(), &parsed);
+      keymetadata_.m_Type=parsed.type;
```

Comment:
Rather than setting individual data members of the KeyMetaData struct here, I suggest adding a setter method in KeyMetaData and calling it here. That way, if the members in KeyMetaData change, you only need to change the signature of the setter and the compiler will tell you any place that you call it with the incorrect member values.